src/analytics/internal/analytics.js

```diff
@@ -0,0 +1,156 @@
+            last_time_used: eventArgs.timestamp,
+            notifType: MapEventTypeToInt[eventArgs.type].notifType,
+        }
+
+        AnalyticsStorage({
+            event: params,
+            statistics: MapEventTypeToInt[eventArgs.type].notifType
+                ? notifParams
+                : undefined,
+        })
+    }
+
+    /**
+     * Load Initial data from dexie for the particular event type
+     * Query to dexie and store into _operationsMap
+     * @param {notifType} type of notif event
+     */
+    async loadInititalData(notifType, isCacheUpdate = false) {
+        await db.transaction('rw', db.tables, async () => {
+            const eventLogCount = await db.eventLog
+                .where('type')
+                .anyOf(MapNotifTypeToIntArray[notifType])
+                .count()
+
+            const eventLogLastTimeUsed = await db.eventLog
+                .where('type')
+                .anyOf(MapNotifTypeToIntArray[notifType])
+                .last()
+
+            // If the data is out of date then we have to update with dexie query and initial loading of data
+            if (this._operationsMap[notifType] === undefined || isCacheUpdate) {
+                // if there is any entry in the dexie then update data of cache
+                if (eventLogLastTimeUsed) {
+                    this._operationsMap[notifType] = {
+                        last_time_used: eventLogLastTimeUsed.timestamp,
+                        value: eventLogCount,
+                    }
+                }
+            } else {
+                if (eventLogLastTimeUsed) {
+                    this._operationsMap[notifType] = {
+                        last_time_used: eventLogLastTimeUsed.timestamp,
+                        value:
+                            eventLogCount +
+                            this._operationsMap[notifType].value,
+                    }
+                }
+            }
+        })
+    }
+
+    incrementvalue(event) {
+        if (!this._operationsMap[event.notifType]) {
+            this._operationsMap[event.notifType] = {
+                value: 1,
+                last_time_used: event.last_time_used,
+            }
+        } else {
+            this._operationsMap[event.notifType] = {
+                value: this._operationsMap[event.notifType].value + 1,
+                last_time_used: event.last_time_used,
+            }
+        }
+    }
+
+    getCountNotif(notifType) {
+        return this._operationsMap[notifType]
+    }
+
+    /**
+     * Track any user-invoked events internally.
+     *
+     * @param {EventTrackInfo} eventArgs
+     */
+    async processEvent(eventArgs) {
+        this._loaded = 1
+        // Prepare the event to store the event in dexie db.
+        const timestamp = Date.now()
+
+        const params = {
+            ...eventArgs,
+            timestamp,
+        }
+
+        // If all the items has been loaded initially
+        if (this._loaded === 1) {
```

Comment:
It's used when we will load the data initially from dexie, so that we have to first register all the operations and we will push into queue and then execute one by one.

Reply:
That's what I presumed it would be for, but that's not the way it's currently implemented. You can look over the 3 current references.

As it's intended to say if the data is loaded or not: 
data loading is going to be async, so rather than setting this to a number it would make more sense to set this as a `Promise`, which you can resolve once that initial data load is done. (then logic that depends on the data to be loaded can `await this._loaded` rather than doing a sync `this._loaded === 1` check, or else you might have `processEvent` calls that do nothing because the data wasn't ready)